src/ORM/Exception/PersistenceFailedException.php

```diff
@@ -47,35 +48,17 @@ public function __construct(EntityInterface $entity, $message, $code = null, $pr
         $this->_entity = $entity;
         if (is_array($message)) {
             $errors = [];
-            foreach ($entity->getErrors() as $field => $error) {
-                $errors[] = $field . ': "' . $this->buildError($error) . '"';
+            foreach (Hash::flatten($entity->getErrors()) as $field => $error) {
+                $errors[] = $field . ': "' . $error . '"';
             }
             if ($errors) {
                 $message[] = implode(', ', $errors);
-                $this->_messageTemplate = 'Entity %s failure (%s).';
+                $this->_messageTemplate = 'Entity %s failure. Found the following errors (%s).';
             }
         }
         parent::__construct($message, $code, $previous);
     }
 
-    /**
-     * @param string|array $error Error message.
-     * @return string
-     */
-    protected function buildError($error)
-    {
-        if (!is_array($error)) {
```

Comment:
We dont need to keep this part? String check?

Reply:
Hash::flatten() always returns a flat key : value pairing.

Reply:
Neat then as it simplies this a lot here :)